src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/MSBuildShellOutNuGetProject.cs

```diff
@@ -326,6 +355,51 @@ public static string GetMSBuildPath(IVsBuildPropertyStorage buildPropertyStorage
             return null;
         }
 
+        public bool IsRestoreRequired(
+            IEnumerable<VersionFolderPathResolver> pathResolvers,
+            ISet<PackageIdentity> packagesChecked,
+            ExternalProjectReferenceContext context)
+        {
+            return true;
```

Comment:
We may add a todo comment to check it in future to see if we really need a restore for CPS based projects as well.